src/irmin-pack/unix/gc.ml

```diff
@@ -268,14 +268,14 @@ module Make (Args : Args) = struct
                 let () = Lwt.wakeup_later t.resolver err in
                 err
           in
-          Lwt.return result
+          result
         in
         if wait then
-          let* status = Async.await t.task in
+          let status = Async.await t.task in
           go status
         else
           match Async.status t.task with
-          | `Running -> Lwt.return_ok `Running
+          | `Running -> Ok `Running
           | status -> go status)
```

Comment:
A solution to the `waitpid`/`lwt` problem would be to turn the existing `finalise` into `finalise_nowait`:
```
val finalise : wait:bool -> t -> ([> 'Running | 'Finalised of stats ], Args.Errs.t) result Lwt.t
val finalise_nowait :       t -> ([> 'Running | 'Finalised of stats ], Args.Errs.t) result
```

and then add a new `finalise_wait` function that uses lwt:
```
val finalise : wait:bool -> t -> ([> 'Running | 'Finalised of stats ], Args.Errs.t) result Lwt.t
val finalise_wait :         t -> ([> 'Running | 'Finalised of stats ], Args.Errs.t) result Lwt.t
```

This way, `lwt` would be gone from all but one function of that part of the code

Reply:
I think this would lead to some code duplication, I'm not sure its worth: its not removing lwt from async and finalise is already the only place in the gc where we return lwt. 

I'm closing this for now. If you disagree, please open this again.